### PR TITLE
Some changes to the vertx impl

### PR DESCRIPTION
**Pull request by adamw**
* build only for 2.12
* format code using scalafmt
* some type safety added

### CODE REVIEW

File: server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/package.scala [hunk 1]
```diff
@@ -15,8 +15,10 @@ package object encoders {
     */
   private[vertx] def inputStreamToBuffer(is: InputStream): Buffer = {
     val buffer = Buffer.buffer()
+    val buf = new Array[Byte](bufferSize)
     while (is.available() > 0) {
-      buffer.appendBytes(is.readNBytes(bufferSize))
+      val read = is.read(buf)
+      buffer.appendBytes(buf, 0, read)
```
Comment (aesteve) [inline server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/package.scala:21]: Are you sure it's gonna work?
Always appending to the start of the buffer ?

Comment (adamw) [reply server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/package.scala:21]: oh ... I though this specifies the slice of `buf` that will be appended

Comment (adamw) [reply server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/package.scala:21]: and as the tests passed ... ;)

Comment (adamw) [reply server/vertx/src/main/scala/sttp/tapir/server/vertx/encoders/package.scala:21]: yes, this seems correct, see the javadocs to the delegating impl (`io.netty.buffer.ByteBuf.writeBytes`) and e.g. the implementaiton in `AbstractByteBuf`:

```
@Override
    public ByteBuf writeBytes(byte[] src) {
        writeBytes(src, 0, src.length);
        return this;
    }
```

**Pull request merged**